back/app/models/idea.rb

```diff
@@ -80,6 +81,18 @@ class Idea < ApplicationRecord
 
   validates_numericality_of :proposed_budget, greater_than_or_equal_to: 0, if: :proposed_budget
 
+  validates :custom_field_values, on: :publication, json: {
+    schema: -> {
+      extra_fields = CustomForm.where(project: project)&.first.custom_fields.find_all { |f| f.code == nil }
```

Comment:
`code` is a string, or `nil`, so this can be simplified:
```suggestion
      extra_fields = CustomForm.where(project: project).first.custom_fields.select(&:code) }
```
`select` and `find_all` are aliases, but `select` expresses intent better: "select all custom fields with a code".
Note that I removed `&` after `CustomForm.where(project: project)`, because `where` returns an ActiveRecord relation. It cannot be `nil`, so `&` is not needed.
If you do not like to use this Ruby pattern, then please replace `== nil` by `.nil?`.

Reply:
Ah right, except here we need to "select custom fields without a code". Removed the useless & and swith to .nil?

Reply:
Oops, yes it should have been "select all custom fields without a code". Then the code should be `extra_fields = CustomForm.where(project: project).first.custom_fields.reject(&:code) `